research-app/src/App.vue

```diff
@@ -461,6 +503,34 @@ export default class App extends WWTAwareComponent {
         this.onMessage(event.data);
       }
     }, false);
+
+    // Handling key presses
+    this._keyboardControlSettings = new KeyboardControlSettings({});
+    this._keyListener = function(e: KeyboardEvent) {
+      const kcs = this._keyboardControlSettings;
+      if (kcs == null || kcs == undefined) { return; }
```

Comment:
Another stylistic thing: I would like to avoid one-line branches in control flow, i.e. `if` statements with their bodies all one one line. It makes it harder to catch the structure of control flow when scanning the code at a high level.